addons/mail/models/mail_tracking_value.py

```diff
@@ -120,11 +116,21 @@ def _create_tracking_values(self, initial_value, new_value, col_name, col_info,
 
     def _tracking_value_format(self):
         """ Return structure and formatted data structure to be used by chatter
-        to display tracking values.
+        to display tracking values. Order it according to asked display, aka
+        ascending sequence (and field name).
 
         :return list: for each tracking value in self, their formatted display
           values given as a dict;
         """
+        if not self:
+            return []
+        field_models = self.field.mapped('model')
+        if len(set(field_models)) > 1:
+            raise ValueError(_('All tracking value should belong to the same model.'))
```

Comment:
I learned recently that error in `ValueError` should not be translated (`ValueError` show a traceback on the client)

so we should either change it to `UserError` (to have a popup on the client) or do not translate the error (if it's considered as a technical error that should be catched during the dev)

Reply:
Untranslated ValueError then :) (as it is a programmation error, not a real error for clients)